File: VB/ImageMemoEdit/ExtendedEditor/MyMemoEdit.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace EditorDescendant

    <System.ComponentModel.DesignerCategory("")>
    <System.ComponentModel.ToolboxItem(True)>
    Public Class MyMemoEdit
        Inherits MemoEdit

'#Region "Service"
        Shared Sub New()
            MyRepositoryItemMemoEdit.Register()
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return MyRepositoryItemMemoEdit.EditorName
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As MyRepositoryItemMemoEdit
            Get
                Return TryCast(MyBase.Properties, MyRepositoryItemMemoEdit)
            End Get
        End Property
'#End Region
    End Class
End Namespace

File: CS/ImageMemoEdit/ExtendedEditor/MyMemoEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace EditorDescendant
{
    [System.ComponentModel.DesignerCategory("")]
    [System.ComponentModel.ToolboxItem(true)]
    public class MyMemoEdit : MemoEdit
    {
        #region Service
        static MyMemoEdit()
        {
            MyRepositoryItemMemoEdit.Register();
        }

        public MyMemoEdit()
            : base()
        {
        }

        public override string EditorTypeName
        { get { return MyRepositoryItemMemoEdit.EditorName; } }
        
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new MyRepositoryItemMemoEdit Properties
        { get { return base.Properties as MyRepositoryItemMemoEdit; } }

     #endregion
    }
}
